plugin/src/main/kotlin/org/gradle/github/dependency/extractor/internal/DependencyExtractorService.kt

```diff
@@ -160,10 +169,13 @@ abstract class DependencyExtractorService :
         }
     }
 
-    fun writeAndGetSnapshotFile(): File =
+    private fun writeAndGetSnapshotFile(): File =
         fileWriter.writeDependencyManifest(gitHubRepositorySnapshotBuilder.build())
 
     override fun close() {
+        if (thrownExceptions.isNotEmpty()) {
+            throw MultipleBuildFailures(thrownExceptions)
+        }
         writeAndGetSnapshotFile()
```

Comment:
From this implementation, I presume that exceptions thrown in `writeAndGetSnapshotFile` are _not_ swallowed by Gradle. Can you confirm?